textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/KeywordPredicate.java

```diff
@@ -28,48 +32,86 @@
     private final List<Attribute> attributeList;
     private final String[] fields;
     private final String query;
-    private final Query queryObject;
+    private final Query luceneQuery;
     private ArrayList<String> tokens;
     private Analyzer analyzer;
+    private IDataStore dataStore;
 
-    public KeywordPredicate(String query, List<Attribute> attributeList, Analyzer analyzer ) throws DataFlowException{
+    public KeywordPredicate(String query, List<Attribute> attributeList, Analyzer analyzer,IDataStore dataStore ) throws DataFlowException{
```

Comment:
@akshaybetala Is `keyword` better name here than `query`? Please change it if you agree.